File: fix-pages.js
const fs = require('fs');
const path = require('path');

// Function to fix common issues in page files
function fixPageFile(filePath) {
  try {
    let content = fs.readFileSync(filePath, 'utf8');
    let modified = false;

    // Fix 'use client' directive issues
    if (content.includes("'use client'") && content.includes('react-helmet-async')) {
      content = content.replace(/import { Helmet } from 'react-helmet-async';\n?/g, '');
      content = content.replace(/<Helmet>[\s\S]*?<\/Helmet>/g, '');
      modified = true;
    }

    // Fix Link components using 'to' instead of 'href'
    if (content.includes('to=')) {
      content = content.replace(/to=/g, 'href=');
      modified = true;
    }

    // Fix generic component names
    if (content.includes('const PagePage: React.FC = () => {')) {
      const fileName = path.basename(path.dirname(filePath));
      const componentName = fileName.split('-').map(word => 
        word.charAt(0).toUpperCase() + word.slice(1)
      ).join('') + 'Page';
      
      content = content.replace(/const PagePage: React.FC = () => {/g, `const ${componentName}: React.FC = () => {`);
      content = content.replace(/export default PagePage;/g, `export default ${componentName};`);
      modified = true;
    }

    // Fix missing icon imports by adding common ones
    const commonIcons = [
      'Send', 'BarChart3', 'Shield', 'Target', 'Zap', 'Globe', 'DollarSign', 
      'Clock', 'Star', 'Activity', 'Lock', 'FileText', 'CreditCard', 
      'Database', 'Building2', 'Sparkles', 'PieChart', 'CheckCircle'
    ];
    
    const existingImports = content.match(/import {[^}]+} from 'lucide-react'/);
    if (existingImports) {
      const currentIcons = existingImports[0].match(/[A-Z][a-zA-Z0-9]+/g) || [];
      const missingIcons = commonIcons.filter(icon => !currentIcons.includes(icon));
      
      if (missingIcons.length > 0) {
        const newImport = `import { ${currentIcons.join(', ')}, ${missingIcons.join(', ')} } from 'lucide-react'`;
        content = content.replace(existingImports[0], newImport);
        modified = true;
      }
    }

    // Fix complex icon objects that might cause serialization issues
    content = content.replace(/icon: <([A-Z][a-zA-Z0-9]+) className="[^"]*" \/>/g, 'icon: $1');
    content = content.replace(/<([A-Z][a-zA-Z0-9]+) className="[^"]*" \/>/g, '<$1 className="w-8 h-8" />');
    
    if (modified) {
      fs.writeFileSync(filePath, content);
      console.log(`Fixed: ${filePath}`);
      return true;
    }
    
    return false;
  } catch (error) {
    console.error(`Error fixing ${filePath}:`, error.message);
    return false;
  }
}

// Function to recursively find and fix all page files
function fixAllPages(dir) {
  const files = fs.readdirSync(dir);
  let fixedCount = 0;

  for (const file of files) {
    const filePath = path.join(dir, file);
    const stat = fs.statSync(filePath);

    if (stat.isDirectory()) {
      // Skip node_modules and .next directories
      if (file !== 'node_modules' && file !== '.next' && !file.startsWith('.')) {
        fixedCount += fixAllPages(filePath);
      }
    } else if (file === 'page.tsx' || file === 'page.js') {
      if (fixPageFile(filePath)) {
        fixedCount++;
      }
    }
  }

  return fixedCount;
}

// Start fixing from the app directory
const appDir = path.join(__dirname, 'app');
console.log('Starting to fix pages...');
const totalFixed = fixAllPages(appDir);
console.log(`Fixed ${totalFixed} page files`);

File: fix_pages.py
#!/usr/bin/env python3
"""
Fix Next.js App Router compatibility issues in page components.
"""

import os
import re
import glob
from pathlib import Path

def fix_page_file(file_path):
    """Fix a single page file to be compatible with Next.js App Router."""
    try:
        with open(file_path, 'r', encoding='utf-8') as f:
            content = f.read()
        
        # Skip if already fixed or not a problematic file
        if 'import Head from' not in content or 'export const metadata' in content:
            return False
        
        # Extract title and description from Head component
        title_match = re.search(r'<title>([^<]+)</title>', content)
        desc_match = re.search(r'<meta name="description" content="([^"]+)"', content)
        
        title = title_match.group(1) if title_match else 'Zion Tech Group'
        description = desc_match.group(1) if desc_match else 'Professional technology services and solutions.'
        
        # Remove 'use client' directive if present
        content = re.sub(r"'use client'\n", '', content)
        
        # Remove Head import
        content = re.sub(r"import Head from 'next/head'\n", '', content)
        
        # Add metadata import and export
        metadata_import = "import { Metadata } from 'next'\n\n"
        metadata_export = f"""export const metadata: Metadata = {{
  title: '{title}',
  description: '{description}',
  robots: 'index, follow',
  openGraph: {{
    type: 'website',
    title: '{title}',
    description: '{description}',
  }},
}}

"""
        
        # Remove Head component and its content
        content = re.sub(r'<Head>.*?</Head>', '', content, flags=re.DOTALL)
        
        # Add metadata import at the top
        content = metadata_import + content
        
        # Add metadata export after imports
        content = re.sub(r'(import.*?\n)(export default)', r'\1' + metadata_export + r'\2', content)
        
        # Write fixed content
        with open(file_path, 'w', encoding='utf-8') as f:
            f.write(content)
        
        print(f"Fixed: {file_path}")
        return True
        
    except Exception as e:
        print(f"Error fixing {file_path}: {e}")
        return False

def main():
    """Fix all page files in the app directory."""
    print("Fixing Next.js App Router compatibility issues...")
    
    # Find all page.tsx files
    page_files = glob.glob('app/**/page.tsx', recursive=True)
    
    fixed_count = 0
    for file_path in page_files:
        if fix_page_file(file_path):
            fixed_count += 1
    
    print(f"Fixed {fixed_count} page files.")

if __name__ == "__main__":
    main()